StarRocks planner use long time 3000 ms.

<!-- (At least include the following, feel free to add if you have more content) -->

### Steps to reproduce the behavior (Required)
1. CREATE TABLE '...'
```
CREATE DATABASE v0;
USE v0;
admin set frontend config ("enable_decimal_v3" = "true");
CREATE TABLE t0 (c_0_0 DATE NULL ,c_0_1 DECIMAL(35, 20)  NOT NULL ,c_0_2 INT NULL ,c_0_3 DATETIME NOT NULL ,c_0_4 TINYINT NOT NULL ,c_0_5 DECIMAL(34, 12)  NOT NULL ,c_0_6 CHAR(31) NULL ,c_0_7 BOOLEAN NULL ,c_0_8 FLOAT REPLACE NULL , INDEX index1_c_0_4(`c_0_4`) USING BITMAP) AGGREGATE KEY (c_0_0,c_0_1,c_0_2,c_0_3,c_0_4,c_0_5,c_0_6,c_0_7) PARTITION BY RANGE(c_0_0) (START ("2010-01-01") END ("2021-12-31") EVERY (INTERVAL 97 day)) DISTRIBUTED BY HASH (c_0_0,c_0_4,c_0_2,c_0_6,c_0_5) BUCKETS 3 properties("replication_num"="3") ;
CREATE TABLE IF NOT EXISTS t1 (c_1_0 DATE NOT NULL,c_1_1 ARRAY<STRING> NOT NULL) PRIMARY KEY (c_1_0) DISTRIBUTED BY HASH (c_1_0) BUCKETS 3 properties("replication_num"="3") ;
DELETE FROM t0 WHERE c_0_1=0.00000490447953788646 AND c_0_2=823279980 AND c_0_3="1969-12-25 22:14:26";
INSERT INTO t1 VALUES ("2010-01-23", ["", "oF", "4u", "JN", "O5bk", "L1o", "G"]), ("2010-01-11", ["JglRSV9K", "479", "", "", "", "n", "R", "V", "U", "", "", "Qdbk9Vi"]);
INSERT INTO t0 VALUES (NULL, 0.00000456701652120433, 7, "2010-01-23 08:00:14", 6, 1872325264.453762283627, "", true, 0.7342448670957158);
INSERT INTO t0 VALUES ("2010-01-02", 0.00000970332763879900, 18, "2010-01-05 08:00:14", 1, 1508374250.404986739222, "r", true, -5.53533237E8);
INSERT INTO t1 VALUES ("2010-01-22", ["", "dPY3pOLv", "Dyh", "l4", "Ny", "Dj", "QQg", "j", "gkBLqPB", "7", "Kg"]), ("2010-01-30", ["dfR", "", "tZ2hufXiJ", "MrIz", "", "", "EumWeAVw", "", "2FjMhiLP", "J", "1", "O", "6QG7ep", "v"]);
CREATE VIEW v0 AS WITH with_t_0 as (SELECT t1_3.c_1_0 FROM t1 AS t1_3) , with_t_1 as (SELECT t1_3.c_1_0 FROM t1 AS t1_3) SELECT t1_3.c_1_0 FROM with_t_1 AS with_t_1,  t1 AS t1_3  LEFT JOIN (SELECT with_t_0.c_1_0 FROM with_t_0 AS with_t_0 ORDER BY with_t_0.c_1_0 DESC) subwith_t_0 ON t1_3.c_1_0 = subwith_t_0.c_1_0 AND subwith_t_0.c_1_0 >= t1_3.c_1_0 AND t1_3.c_1_0 != subwith_t_0.c_1_0 AND t1_3.c_1_0 != subwith_t_0.c_1_0;
CREATE VIEW v1 AS WITH with_t_0 as (SELECT t1_3.c_1_0 FROM t1 AS t1_3) , with_t_1 as (SELECT t1_3.c_1_0 FROM t1 AS t1_3) SELECT t1_3.c_1_0 FROM with_t_1 AS with_t_1,  t1 AS t1_3  LEFT JOIN (SELECT with_t_0.c_1_0 FROM with_t_0 AS with_t_0 ORDER BY with_t_0.c_1_0 DESC) subwith_t_0 ON t1_3.c_1_0 = subwith_t_0.c_1_0 AND subwith_t_0.c_1_0 >= t1_3.c_1_0 AND t1_3.c_1_0 != subwith_t_0.c_1_0 AND t1_3.c_1_0 != subwith_t_0.c_1_0 WHERE ((subwith_t_0.c_1_0) IS NULL);
CREATE VIEW v2 AS WITH with_t_0 as (SELECT t1_3.c_1_0 FROM t1 AS t1_3) , with_t_1 as (SELECT t1_3.c_1_0 FROM t1 AS t1_3) SELECT t1_3.c_1_0 FROM with_t_1 AS with_t_1,  t1 AS t1_3  LEFT JOIN (SELECT with_t_0.c_1_0 FROM with_t_0 AS with_t_0 ORDER BY with_t_0.c_1_0 DESC) subwith_t_0 ON t1_3.c_1_0 = subwith_t_0.c_1_0 AND subwith_t_0.c_1_0 >= t1_3.c_1_0 AND t1_3.c_1_0 != subwith_t_0.c_1_0 AND t1_3.c_1_0 != subwith_t_0.c_1_0 WHERE (NOT (((subwith_t_0.c_1_0) IS NULL)));
CREATE VIEW v3 AS WITH with_t_0 as (SELECT t1_3.c_1_0 FROM t1 AS t1_3) , with_t_1 as (SELECT t1_3.c_1_0 FROM t1 AS t1_3) SELECT t1_3.c_1_0 FROM with_t_1 AS with_t_1,  t1 AS t1_3  LEFT JOIN (SELECT with_t_0.c_1_0 FROM with_t_0 AS with_t_0 ORDER BY with_t_0.c_1_0 DESC) subwith_t_0 ON t1_3.c_1_0 = subwith_t_0.c_1_0 AND subwith_t_0.c_1_0 >= t1_3.c_1_0 AND t1_3.c_1_0 != subwith_t_0.c_1_0 AND t1_3.c_1_0 != subwith_t_0.c_1_0 WHERE ((((subwith_t_0.c_1_0) IS NULL)) IS NULL);
CREATE VIEW v4 AS WITH with_t_0 as (SELECT t1_3.c_1_0 FROM t1 AS t1_3) , with_t_1 as (SELECT t1_3.c_1_0 FROM t1 AS t1_3) SELECT t1_3.c_1_0 FROM with_t_1 AS with_t_1,  t1 AS t1_3  LEFT JOIN (SELECT with_t_0.c_1_0 FROM with_t_0 AS with_t_0 ORDER BY with_t_0.c_1_0 DESC) subwith_t_0 ON t1_3.c_1_0 = subwith_t_0.c_1_0 AND subwith_t_0.c_1_0 >= t1_3.c_1_0 AND t1_3.c_1_0 != subwith_t_0.c_1_0 AND t1_3.c_1_0 != subwith_t_0.c_1_0 WHERE ((subwith_t_0.c_1_0) IS NULL) UNION ALL SELECT t1_3.c_1_0 FROM with_t_1 AS with_t_1,  t1 AS t1_3  LEFT JOIN (SELECT with_t_0.c_1_0 FROM with_t_0 AS with_t_0 ORDER BY with_t_0.c_1_0 DESC) subwith_t_0 ON t1_3.c_1_0 = subwith_t_0.c_1_0 AND subwith_t_0.c_1_0 >= t1_3.c_1_0 AND t1_3.c_1_0 != subwith_t_0.c_1_0 AND t1_3.c_1_0 != subwith_t_0.c_1_0 WHERE (NOT (((subwith_t_0.c_1_0) IS NULL))) UNION ALL SELECT t1_3.c_1_0 FROM with_t_1 AS with_t_1,  t1 AS t1_3  LEFT JOIN (SELECT with_t_0.c_1_0 FROM with_t_0 AS with_t_0 ORDER BY with_t_0.c_1_0 DESC) subwith_t_0 ON t1_3.c_1_0 = subwith_t_0.c_1_0 AND subwith_t_0.c_1_0 >= t1_3.c_1_0 AND t1_3.c_1_0 != subwith_t_0.c_1_0 AND t1_3.c_1_0 != subwith_t_0.c_1_0 WHERE ((((subwith_t_0.c_1_0) IS NULL)) IS NULL);
CREATE VIEW v5 AS WITH with_t_0 as (SELECT t1_23.c_1_0 FROM t1 AS t1_23 WHERE true) , with_t_1 as (SELECT v0_24.c_1_0 FROM v0 AS v0_24 WHERE true) SELECT v0_24.c_1_0 FROM with_t_0 AS with_t_0, with_t_1 AS with_t_1,  v0 AS v0_24 LEFT SEMI JOIN (SELECT t1_23.c_1_0 FROM t1 AS t1_23 WHERE false) subt1 ON v0_24.c_1_0 = subt1.c_1_0 AND v0_24.c_1_0 >= subt1.c_1_0;
CREATE VIEW v6 AS WITH with_t_0 as (SELECT t1_23.c_1_0 FROM t1 AS t1_23 WHERE true) , with_t_1 as (SELECT v0_24.c_1_0 FROM v0 AS v0_24 WHERE true) SELECT v0_24.c_1_0 FROM with_t_0 AS with_t_0, with_t_1 AS with_t_1,  v0 AS v0_24 LEFT SEMI JOIN (SELECT t1_23.c_1_0 FROM t1 AS t1_23 WHERE false) subt1 ON v0_24.c_1_0 = subt1.c_1_0 AND v0_24.c_1_0 >= subt1.c_1_0 WHERE (NOT ((NOT (true))));
CREATE VIEW v36 AS WITH with_t_0 as (SELECT t1_23.c_1_0 FROM t1 AS t1_23 WHERE true) , with_t_1 as (SELECT v0_24.c_1_0 FROM v0 AS v0_24 WHERE true) SELECT v0_24.c_1_0 FROM with_t_0 AS with_t_0, with_t_1 AS with_t_1,  v0 AS v0_24 LEFT SEMI JOIN (SELECT t1_23.c_1_0 FROM t1 AS t1_23 WHERE false) subt1 ON v0_24.c_1_0 = subt1.c_1_0 AND v0_24.c_1_0 >= subt1.c_1_0 WHERE (NOT ((NOT ((NOT (true))))));
CREATE VIEW v7 AS WITH with_t_0 as (SELECT t1_23.c_1_0 FROM t1 AS t1_23 WHERE true) , with_t_1 as (SELECT v0_24.c_1_0 FROM v0 AS v0_24 WHERE true) SELECT v0_24.c_1_0 FROM with_t_0 AS with_t_0, with_t_1 AS with_t_1,  v0 AS v0_24 LEFT SEMI JOIN (SELECT t1_23.c_1_0 FROM t1 AS t1_23 WHERE false) subt1 ON v0_24.c_1_0 = subt1.c_1_0 AND v0_24.c_1_0 >= subt1.c_1_0 WHERE (((NOT ((NOT (true))))) IS NULL);
CREATE VIEW v8 AS WITH with_t_0 as (SELECT t1_23.c_1_0 FROM t1 AS t1_23 WHERE true) , with_t_1 as (SELECT v0_24.c_1_0 FROM v0 AS v0_24 WHERE true) SELECT v0_24.c_1_0 FROM with_t_0 AS with_t_0, with_t_1 AS with_t_1,  v0 AS v0_24 LEFT SEMI JOIN (SELECT t1_23.c_1_0 FROM t1 AS t1_23 WHERE false) subt1 ON v0_24.c_1_0 = subt1.c_1_0 AND v0_24.c_1_0 >= subt1.c_1_0 WHERE (NOT ((NOT (true)))) UNION ALL SELECT v0_24.c_1_0 FROM with_t_0 AS with_t_0, with_t_1 AS with_t_1,  v0 AS v0_24 LEFT SEMI JOIN (SELECT t1_23.c_1_0 FROM t1 AS t1_23 WHERE false) subt1 ON v0_24.c_1_0 = subt1.c_1_0 AND v0_24.c_1_0 >= subt1.c_1_0 WHERE (NOT ((NOT ((NOT (true)))))) UNION ALL SELECT v0_24.c_1_0 FROM with_t_0 AS with_t_0, with_t_1 AS with_t_1,  v0 AS v0_24 LEFT SEMI JOIN (SELECT t1_23.c_1_0 FROM t1 AS t1_23 WHERE false) subt1 ON v0_24.c_1_0 = subt1.c_1_0 AND v0_24.c_1_0 >= subt1.c_1_0 WHERE (((NOT ((NOT (true))))) IS NULL);
CREATE VIEW v9 AS SELECT subv36.c_1_0 FROM v3 AS v3_72, t0 AS t0_67, t1 AS t1_68, v0 AS v0_69, v2 AS v2_71,  (SELECT v5_75.c_1_0 FROM v5 AS v5_75) subv5 RIGHT SEMI JOIN (SELECT v36_73.c_1_0 FROM v36 AS v36_73 ORDER BY 823279980) subv36 ON subv5.c_1_0 = subv36.c_1_0 AND subv5.c_1_0 < CASE  WHEN true THEN subv36.c_1_0  WHEN false THEN subv36.c_1_0  WHEN true THEN subv36.c_1_0 END AND subv5.c_1_0 != subv36.c_1_0 AND MICROSECONDS_SUB("1970-01-10 20:41:19", -1397228483)  >= subv36.c_1_0,  (SELECT v7_77.c_1_0 FROM v7 AS v7_77) subv7  LEFT JOIN v4 AS v4_74 ON subv7.c_1_0 = v4_74.c_1_0 AND subv7.c_1_0 <= v4_74.c_1_0 AND subv7.c_1_0 != v4_74.c_1_0 AND subv7.c_1_0 >= v4_74.c_1_0,  v1 AS v1_70 FULL JOIN v8 AS v8_78 ON v1_70.c_1_0 = v8_78.c_1_0 AND v1_70.c_1_0 != v8_78.c_1_0 AND v1_70.c_1_0 >= v8_78.c_1_0 AND v1_70.c_1_0 = v8_78.c_1_0;
CREATE VIEW v10 AS SELECT subv36.c_1_0 FROM v3 AS v3_72, t0 AS t0_67, t1 AS t1_68, v0 AS v0_69, v2 AS v2_71,  (SELECT v5_75.c_1_0 FROM v5 AS v5_75) subv5 RIGHT SEMI JOIN (SELECT v36_73.c_1_0 FROM v36 AS v36_73 ORDER BY 823279980) subv36 ON subv5.c_1_0 = subv36.c_1_0 AND subv5.c_1_0 < CASE  WHEN true THEN subv36.c_1_0  WHEN false THEN subv36.c_1_0  WHEN true THEN subv36.c_1_0 END AND subv5.c_1_0 != subv36.c_1_0 AND MICROSECONDS_SUB("1970-01-10 20:41:19", -1397228483)  >= subv36.c_1_0,  (SELECT v7_77.c_1_0 FROM v7 AS v7_77) subv7  LEFT JOIN v4 AS v4_74 ON subv7.c_1_0 = v4_74.c_1_0 AND subv7.c_1_0 <= v4_74.c_1_0 AND subv7.c_1_0 != v4_74.c_1_0 AND subv7.c_1_0 >= v4_74.c_1_0,  v1 AS v1_70 FULL JOIN v8 AS v8_78 ON v1_70.c_1_0 = v8_78.c_1_0 AND v1_70.c_1_0 != v8_78.c_1_0 AND v1_70.c_1_0 >= v8_78.c_1_0 AND v1_70.c_1_0 = v8_78.c_1_0 WHERE true;
CREATE VIEW v11 AS SELECT subv36.c_1_0 FROM v3 AS v3_72, t0 AS t0_67, t1 AS t1_68, v0 AS v0_69, v2 AS v2_71,  (SELECT v5_75.c_1_0 FROM v5 AS v5_75) subv5 RIGHT SEMI JOIN (SELECT v36_73.c_1_0 FROM v36 AS v36_73 ORDER BY 823279980) subv36 ON subv5.c_1_0 = subv36.c_1_0 AND subv5.c_1_0 < CASE  WHEN true THEN subv36.c_1_0  WHEN false THEN subv36.c_1_0  WHEN true THEN subv36.c_1_0 END AND subv5.c_1_0 != subv36.c_1_0 AND MICROSECONDS_SUB("1970-01-10 20:41:19", -1397228483)  >= subv36.c_1_0,  (SELECT v7_77.c_1_0 FROM v7 AS v7_77) subv7  LEFT JOIN v4 AS v4_74 ON subv7.c_1_0 = v4_74.c_1_0 AND subv7.c_1_0 <= v4_74.c_1_0 AND subv7.c_1_0 != v4_74.c_1_0 AND subv7.c_1_0 >= v4_74.c_1_0,  v1 AS v1_70 FULL JOIN v8 AS v8_78 ON v1_70.c_1_0 = v8_78.c_1_0 AND v1_70.c_1_0 != v8_78.c_1_0 AND v1_70.c_1_0 >= v8_78.c_1_0 AND v1_70.c_1_0 = v8_78.c_1_0 WHERE (NOT (true));
CREATE VIEW v12 AS SELECT subv36.c_1_0 FROM v3 AS v3_72, t0 AS t0_67, t1 AS t1_68, v0 AS v0_69, v2 AS v2_71,  (SELECT v5_75.c_1_0 FROM v5 AS v5_75) subv5 RIGHT SEMI JOIN (SELECT v36_73.c_1_0 FROM v36 AS v36_73 ORDER BY 823279980) subv36 ON subv5.c_1_0 = subv36.c_1_0 AND subv5.c_1_0 < CASE  WHEN true THEN subv36.c_1_0  WHEN false THEN subv36.c_1_0  WHEN true THEN subv36.c_1_0 END AND subv5.c_1_0 != subv36.c_1_0 AND MICROSECONDS_SUB("1970-01-10 20:41:19", -1397228483)  >= subv36.c_1_0,  (SELECT v7_77.c_1_0 FROM v7 AS v7_77) subv7  LEFT JOIN v4 AS v4_74 ON subv7.c_1_0 = v4_74.c_1_0 AND subv7.c_1_0 <= v4_74.c_1_0 AND subv7.c_1_0 != v4_74.c_1_0 AND subv7.c_1_0 >= v4_74.c_1_0,  v1 AS v1_70 FULL JOIN v8 AS v8_78 ON v1_70.c_1_0 = v8_78.c_1_0 AND v1_70.c_1_0 != v8_78.c_1_0 AND v1_70.c_1_0 >= v8_78.c_1_0 AND v1_70.c_1_0 = v8_78.c_1_0 WHERE ((true) IS NULL);
CREATE VIEW v13 AS SELECT subv36.c_1_0 FROM v3 AS v3_72, t0 AS t0_67, t1 AS t1_68, v0 AS v0_69, v2 AS v2_71,  (SELECT v5_75.c_1_0 FROM v5 AS v5_75) subv5 RIGHT SEMI JOIN (SELECT v36_73.c_1_0 FROM v36 AS v36_73 ORDER BY 823279980) subv36 ON subv5.c_1_0 = subv36.c_1_0 AND subv5.c_1_0 < CASE  WHEN true THEN subv36.c_1_0  WHEN false THEN subv36.c_1_0  WHEN true THEN subv36.c_1_0 END AND subv5.c_1_0 != subv36.c_1_0 AND MICROSECONDS_SUB("1970-01-10 20:41:19", -1397228483)  >= subv36.c_1_0,  (SELECT v7_77.c_1_0 FROM v7 AS v7_77) subv7  LEFT JOIN v4 AS v4_74 ON subv7.c_1_0 = v4_74.c_1_0 AND subv7.c_1_0 <= v4_74.c_1_0 AND subv7.c_1_0 != v4_74.c_1_0 AND subv7.c_1_0 >= v4_74.c_1_0,  v1 AS v1_70 FULL JOIN v8 AS v8_78 ON v1_70.c_1_0 = v8_78.c_1_0 AND v1_70.c_1_0 != v8_78.c_1_0 AND v1_70.c_1_0 >= v8_78.c_1_0 AND v1_70.c_1_0 = v8_78.c_1_0 WHERE true UNION ALL SELECT subv36.c_1_0 FROM v3 AS v3_72, t0 AS t0_67, t1 AS t1_68, v0 AS v0_69, v2 AS v2_71,  (SELECT v5_75.c_1_0 FROM v5 AS v5_75) subv5 RIGHT SEMI JOIN (SELECT v36_73.c_1_0 FROM v36 AS v36_73 ORDER BY 823279980) subv36 ON subv5.c_1_0 = subv36.c_1_0 AND subv5.c_1_0 < CASE  WHEN true THEN subv36.c_1_0  WHEN false THEN subv36.c_1_0  WHEN true THEN subv36.c_1_0 END AND subv5.c_1_0 != subv36.c_1_0 AND MICROSECONDS_SUB("1970-01-10 20:41:19", -1397228483)  >= subv36.c_1_0,  (SELECT v7_77.c_1_0 FROM v7 AS v7_77) subv7  LEFT JOIN v4 AS v4_74 ON subv7.c_1_0 = v4_74.c_1_0 AND subv7.c_1_0 <= v4_74.c_1_0 AND subv7.c_1_0 != v4_74.c_1_0 AND subv7.c_1_0 >= v4_74.c_1_0,  v1 AS v1_70 FULL JOIN v8 AS v8_78 ON v1_70.c_1_0 = v8_78.c_1_0 AND v1_70.c_1_0 != v8_78.c_1_0 AND v1_70.c_1_0 >= v8_78.c_1_0 AND v1_70.c_1_0 = v8_78.c_1_0 WHERE (NOT (true)) UNION ALL SELECT subv36.c_1_0 FROM v3 AS v3_72, t0 AS t0_67, t1 AS t1_68, v0 AS v0_69, v2 AS v2_71,  (SELECT v5_75.c_1_0 FROM v5 AS v5_75) subv5 RIGHT SEMI JOIN (SELECT v36_73.c_1_0 FROM v36 AS v36_73 ORDER BY 823279980) subv36 ON subv5.c_1_0 = subv36.c_1_0 AND subv5.c_1_0 < CASE  WHEN true THEN subv36.c_1_0  WHEN false THEN subv36.c_1_0  WHEN true THEN subv36.c_1_0 END AND subv5.c_1_0 != subv36.c_1_0 AND MICROSECONDS_SUB("1970-01-10 20:41:19", -1397228483)  >= subv36.c_1_0,  (SELECT v7_77.c_1_0 FROM v7 AS v7_77) subv7  LEFT JOIN v4 AS v4_74 ON subv7.c_1_0 = v4_74.c_1_0 AND subv7.c_1_0 <= v4_74.c_1_0 AND subv7.c_1_0 != v4_74.c_1_0 AND subv7.c_1_0 >= v4_74.c_1_0,  v1 AS v1_70 FULL JOIN v8 AS v8_78 ON v1_70.c_1_0 = v8_78.c_1_0 AND v1_70.c_1_0 != v8_78.c_1_0 AND v1_70.c_1_0 >= v8_78.c_1_0 AND v1_70.c_1_0 = v8_78.c_1_0 WHERE ((true) IS NULL);
CREATE VIEW v14 AS SELECT subv0.c_1_0 FROM t1 AS t1_138, v7 AS v7_151, t0 AS t0_137, v12 AS v12_143, v3 AS v3_146,  (SELECT v0_139.c_1_0 FROM v0 AS v0_139 ORDER BY v0_139.c_1_0 DESC) subv0 LEFT ANTI JOIN (SELECT v6_150.c_1_0 FROM v6 AS v6_150) subv6 ON subv0.c_1_0 = subv6.c_1_0 AND CAST(subv0.c_1_0 AS DATE )  = subv6.c_1_0,  v36 AS v36_147 RIGHT SEMI JOIN v4 AS v4_148 ON v36_147.c_1_0 = v4_148.c_1_0 AND v36_147.c_1_0 > v4_148.c_1_0 AND v36_147.c_1_0 = v4_148.c_1_0 AND v36_147.c_1_0 < v4_148.c_1_0,  v11 AS v11_142 RIGHT ANTI JOIN v5 AS v5_149 ON v11_142.c_1_0 = v5_149.c_1_0 AND v11_142.c_1_0 >= v5_149.c_1_0 AND v11_142.c_1_0 < v5_149.c_1_0 AND CAST(v5_149.c_1_0 AS DATE )  < "1969-12-10";
CREATE VIEW v15 AS SELECT subv0.c_1_0 FROM t1 AS t1_138, v7 AS v7_151, t0 AS t0_137, v12 AS v12_143, v3 AS v3_146,  (SELECT v0_139.c_1_0 FROM v0 AS v0_139 ORDER BY v0_139.c_1_0 DESC) subv0 LEFT ANTI JOIN (SELECT v6_150.c_1_0 FROM v6 AS v6_150) subv6 ON subv0.c_1_0 = subv6.c_1_0 AND CAST(subv0.c_1_0 AS DATE )  = subv6.c_1_0,  v36 AS v36_147 RIGHT SEMI JOIN v4 AS v4_148 ON v36_147.c_1_0 = v4_148.c_1_0 AND v36_147.c_1_0 > v4_148.c_1_0 AND v36_147.c_1_0 = v4_148.c_1_0 AND v36_147.c_1_0 < v4_148.c_1_0,  v11 AS v11_142 RIGHT ANTI JOIN v5 AS v5_149 ON v11_142.c_1_0 = v5_149.c_1_0 AND v11_142.c_1_0 >= v5_149.c_1_0 AND v11_142.c_1_0 < v5_149.c_1_0 AND CAST(v5_149.c_1_0 AS DATE )  < "1969-12-10" WHERE  ( v4_148.c_1_0 ) <= ( v4_148.c_1_0 ) ;
CREATE VIEW v16 AS SELECT subv0.c_1_0 FROM t1 AS t1_138, v7 AS v7_151, t0 AS t0_137, v12 AS v12_143, v3 AS v3_146,  (SELECT v0_139.c_1_0 FROM v0 AS v0_139 ORDER BY v0_139.c_1_0 DESC) subv0 LEFT ANTI JOIN (SELECT v6_150.c_1_0 FROM v6 AS v6_150) subv6 ON subv0.c_1_0 = subv6.c_1_0 AND CAST(subv0.c_1_0 AS DATE )  = subv6.c_1_0,  v36 AS v36_147 RIGHT SEMI JOIN v4 AS v4_148 ON v36_147.c_1_0 = v4_148.c_1_0 AND v36_147.c_1_0 > v4_148.c_1_0 AND v36_147.c_1_0 = v4_148.c_1_0 AND v36_147.c_1_0 < v4_148.c_1_0,  v11 AS v11_142 RIGHT ANTI JOIN v5 AS v5_149 ON v11_142.c_1_0 = v5_149.c_1_0 AND v11_142.c_1_0 >= v5_149.c_1_0 AND v11_142.c_1_0 < v5_149.c_1_0 AND CAST(v5_149.c_1_0 AS DATE )  < "1969-12-10" WHERE (NOT ( ( v4_148.c_1_0 ) <= ( v4_148.c_1_0 ) ));
CREATE VIEW v17 AS SELECT subv0.c_1_0 FROM t1 AS t1_138, v7 AS v7_151, t0 AS t0_137, v12 AS v12_143, v3 AS v3_146,  (SELECT v0_139.c_1_0 FROM v0 AS v0_139 ORDER BY v0_139.c_1_0 DESC) subv0 LEFT ANTI JOIN (SELECT v6_150.c_1_0 FROM v6 AS v6_150) subv6 ON subv0.c_1_0 = subv6.c_1_0 AND CAST(subv0.c_1_0 AS DATE )  = subv6.c_1_0,  v36 AS v36_147 RIGHT SEMI JOIN v4 AS v4_148 ON v36_147.c_1_0 = v4_148.c_1_0 AND v36_147.c_1_0 > v4_148.c_1_0 AND v36_147.c_1_0 = v4_148.c_1_0 AND v36_147.c_1_0 < v4_148.c_1_0,  v11 AS v11_142 RIGHT ANTI JOIN v5 AS v5_149 ON v11_142.c_1_0 = v5_149.c_1_0 AND v11_142.c_1_0 >= v5_149.c_1_0 AND v11_142.c_1_0 < v5_149.c_1_0 AND CAST(v5_149.c_1_0 AS DATE )  < "1969-12-10" WHERE (( ( v4_148.c_1_0 ) <= ( v4_148.c_1_0 ) ) IS NULL);
CREATE VIEW v18 AS SELECT subv0.c_1_0 FROM t1 AS t1_138, v7 AS v7_151, t0 AS t0_137, v12 AS v12_143, v3 AS v3_146,  (SELECT v0_139.c_1_0 FROM v0 AS v0_139 ORDER BY v0_139.c_1_0 DESC) subv0 LEFT ANTI JOIN (SELECT v6_150.c_1_0 FROM v6 AS v6_150) subv6 ON subv0.c_1_0 = subv6.c_1_0 AND CAST(subv0.c_1_0 AS DATE )  = subv6.c_1_0,  v36 AS v36_147 RIGHT SEMI JOIN v4 AS v4_148 ON v36_147.c_1_0 = v4_148.c_1_0 AND v36_147.c_1_0 > v4_148.c_1_0 AND v36_147.c_1_0 = v4_148.c_1_0 AND v36_147.c_1_0 < v4_148.c_1_0,  v11 AS v11_142 RIGHT ANTI JOIN v5 AS v5_149 ON v11_142.c_1_0 = v5_149.c_1_0 AND v11_142.c_1_0 >= v5_149.c_1_0 AND v11_142.c_1_0 < v5_149.c_1_0 AND CAST(v5_149.c_1_0 AS DATE )  < "1969-12-10" WHERE  ( v4_148.c_1_0 ) <= ( v4_148.c_1_0 )  UNION ALL SELECT subv0.c_1_0 FROM t1 AS t1_138, v7 AS v7_151, t0 AS t0_137, v12 AS v12_143, v3 AS v3_146,  (SELECT v0_139.c_1_0 FROM v0 AS v0_139 ORDER BY v0_139.c_1_0 DESC) subv0 LEFT ANTI JOIN (SELECT v6_150.c_1_0 FROM v6 AS v6_150) subv6 ON subv0.c_1_0 = subv6.c_1_0 AND CAST(subv0.c_1_0 AS DATE )  = subv6.c_1_0,  v36 AS v36_147 RIGHT SEMI JOIN v4 AS v4_148 ON v36_147.c_1_0 = v4_148.c_1_0 AND v36_147.c_1_0 > v4_148.c_1_0 AND v36_147.c_1_0 = v4_148.c_1_0 AND v36_147.c_1_0 < v4_148.c_1_0,  v11 AS v11_142 RIGHT ANTI JOIN v5 AS v5_149 ON v11_142.c_1_0 = v5_149.c_1_0 AND v11_142.c_1_0 >= v5_149.c_1_0 AND v11_142.c_1_0 < v5_149.c_1_0 AND CAST(v5_149.c_1_0 AS DATE )  < "1969-12-10" WHERE (NOT ( ( v4_148.c_1_0 ) <= ( v4_148.c_1_0 ) )) UNION ALL SELECT subv0.c_1_0 FROM t1 AS t1_138, v7 AS v7_151, t0 AS t0_137, v12 AS v12_143, v3 AS v3_146,  (SELECT v0_139.c_1_0 FROM v0 AS v0_139 ORDER BY v0_139.c_1_0 DESC) subv0 LEFT ANTI JOIN (SELECT v6_150.c_1_0 FROM v6 AS v6_150) subv6 ON subv0.c_1_0 = subv6.c_1_0 AND CAST(subv0.c_1_0 AS DATE )  = subv6.c_1_0,  v36 AS v36_147 RIGHT SEMI JOIN v4 AS v4_148 ON v36_147.c_1_0 = v4_148.c_1_0 AND v36_147.c_1_0 > v4_148.c_1_0 AND v36_147.c_1_0 = v4_148.c_1_0 AND v36_147.c_1_0 < v4_148.c_1_0,  v11 AS v11_142 RIGHT ANTI JOIN v5 AS v5_149 ON v11_142.c_1_0 = v5_149.c_1_0 AND v11_142.c_1_0 >= v5_149.c_1_0 AND v11_142.c_1_0 < v5_149.c_1_0 AND CAST(v5_149.c_1_0 AS DATE )  < "1969-12-10" WHERE (( ( v4_148.c_1_0 ) <= ( v4_148.c_1_0 ) ) IS NULL);
CREATE VIEW v19 AS SELECT subv6.c_1_0 FROM v9 AS v9_253, v7 AS v7_251, v18 AS v18_244, v11 AS v11_237, v4 AS v4_248, t1 AS t1_233, v5 AS v5_249,  v0 AS v0_234 RIGHT ANTI JOIN (SELECT v6_250.c_1_0 FROM v6 AS v6_250 WHERE CAST("1969-12-10" AS BOOLEAN ) ) subv6 ON v0_234.c_1_0 = subv6.c_1_0 AND v0_234.c_1_0 < subv6.c_1_0 AND v0_234.c_1_0 >= subv6.c_1_0 AND v0_234.c_1_0 < subv6.c_1_0;
CREATE VIEW v20 AS SELECT subv6.c_1_0 FROM v9 AS v9_253, v7 AS v7_251, v18 AS v18_244, v11 AS v11_237, v4 AS v4_248, t1 AS t1_233, v5 AS v5_249,  v0 AS v0_234 RIGHT ANTI JOIN (SELECT v6_250.c_1_0 FROM v6 AS v6_250 WHERE CAST("1969-12-10" AS BOOLEAN ) ) subv6 ON v0_234.c_1_0 = subv6.c_1_0 AND v0_234.c_1_0 < subv6.c_1_0 AND v0_234.c_1_0 >= subv6.c_1_0 AND v0_234.c_1_0 < subv6.c_1_0 WHERE false;
CREATE VIEW v21 AS SELECT subv6.c_1_0 FROM v9 AS v9_253, v7 AS v7_251, v18 AS v18_244, v11 AS v11_237, v4 AS v4_248, t1 AS t1_233, v5 AS v5_249,  v0 AS v0_234 RIGHT ANTI JOIN (SELECT v6_250.c_1_0 FROM v6 AS v6_250 WHERE CAST("1969-12-10" AS BOOLEAN ) ) subv6 ON v0_234.c_1_0 = subv6.c_1_0 AND v0_234.c_1_0 < subv6.c_1_0 AND v0_234.c_1_0 >= subv6.c_1_0 AND v0_234.c_1_0 < subv6.c_1_0 WHERE (NOT (false));
CREATE VIEW v22 AS SELECT subv6.c_1_0 FROM v9 AS v9_253, v7 AS v7_251, v18 AS v18_244, v11 AS v11_237, v4 AS v4_248, t1 AS t1_233, v5 AS v5_249,  v0 AS v0_234 RIGHT ANTI JOIN (SELECT v6_250.c_1_0 FROM v6 AS v6_250 WHERE CAST("1969-12-10" AS BOOLEAN ) ) subv6 ON v0_234.c_1_0 = subv6.c_1_0 AND v0_234.c_1_0 < subv6.c_1_0 AND v0_234.c_1_0 >= subv6.c_1_0 AND v0_234.c_1_0 < subv6.c_1_0 WHERE ((false) IS NULL);
```
2. INSERT INTO '....'
3. SELECT  '....'
```
SELECT * FROM v20 UNION ALL SELECT * FROM v21 UNION ALL SELECT * FROM v22;
```
### Expected behavior (Required)

### Real behavior (Required)
```
mysql> SELECT * FROM v20 UNION ALL SELECT * FROM v21 UNION ALL SELECT * FROM v22;
ERROR 1064 (HY000): StarRocks planner use long time 3000 ms, This probably because 1. FE Full GC, 2. Hive external table fetch metadata took a long time, 3. The SQL is very complex. You could 1. adjust FE JVM config, 2. try query again, 3. enlarge new_planner_optimize_timeout session variable
```
### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`
 - BRANCH-2.2-RELEASE dffad40

--java.lang.AssertionError: SELECT * FROM v20 UNION ALL SELECT * FROM v44 UNION ALL SELECT * FROM v21;
--	at sqlancer.common.query.QueryAdapter.checkException(QueryAdapter.java:98)
--	at sqlancer.common.query.QueryAdapter.executeAndGet(QueryAdapter.java:128)
--	at sqlancer.ComparatorHelper.getResultSetFirstColumnAsString(ComparatorHelper.java:54)
--	at sqlancer.ComparatorHelper.getCombinedResultSet(ComparatorHelper.java:138)
--	at sqlancer.starrocks.oracle.tlp.StarRocksTLPViewOracle.checkViewOnSingleUnion(StarRocksTLPViewOracle.java:68)
--	at sqlancer.starrocks.oracle.tlp.StarRocksTLPViewOracle.check(StarRocksTLPViewOracle.java:42)
--	at sqlancer.ProviderAdapter.generateAndTestDatabase(ProviderAdapter.java:55)
--	at sqlancer.Main$DBMSExecutor.run(Main.java:326)
--	at sqlancer.Main$2.run(Main.java:521)
--	at sqlancer.Main$2.runThread(Main.java:498)
--	at sqlancer.Main$2.run(Main.java:488)
--	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
--	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
--	at java.lang.Thread.run(Thread.java:748)
--Caused by: com.mysql.jdbc.exceptions.jdbc4.MySQLSyntaxErrorException: StarRocks planner use long time 3000 ms, This probably because 1. FE Full GC, 2. Hive external table fetch metadata took a long time, 3. The SQL is very complex. You could 1. adjust FE JVM config, 2. try query again, 3. enlarge new_planner_optimize_timeout session variable
--	at sun.reflect.GeneratedConstructorAccessor5.newInstance(Unknown Source)
--	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
--	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
--	at com.mysql.jdbc.Util.handleNewInstance(Util.java:425)
--	at com.mysql.jdbc.Util.getInstance(Util.java:408)
--	at com.mysql.jdbc.SQLError.createSQLException(SQLError.java:943)
--	at com.mysql.jdbc.MysqlIO.checkErrorPacket(MysqlIO.java:3973)
--	at com.mysql.jdbc.MysqlIO.checkErrorPacket(MysqlIO.java:3909)
--	at com.mysql.jdbc.MysqlIO.sendCommand(MysqlIO.java:2527)
--	at com.mysql.jdbc.MysqlIO.sqlQueryDirect(MysqlIO.java:2680)
--	at com.mysql.jdbc.ConnectionImpl.execSQL(ConnectionImpl.java:2497)
--	at com.mysql.jdbc.ConnectionImpl.execSQL(ConnectionImpl.java:2455)
--	at com.mysql.jdbc.StatementImpl.executeQuery(StatementImpl.java:1369)
--	at sqlancer.common.query.QueryAdapter.executeAndGet(QueryAdapter.java:118)
--	... 12 more
---- Time: 2022/06/14 23:59:28
-- Database: v18
-- Database version: 5.1.0
-- seed value: 1655222268655
DROP DATABASE IF EXISTS v18;
CREATE DATABASE v18;
USE v18;
admin set frontend config ("enable_decimal_v3" = "true");
CREATE TABLE t0 (c_0_0 DATETIME NULL ) AGGREGATE KEY (c_0_0) DISTRIBUTED BY HASH (c_0_0) BUCKETS 3 properties("replication_num"="3") ;
CREATE TABLE t1 (c_1_0 DATETIME NULL ) DUPLICATE KEY (c_1_0) PARTITION BY RANGE(c_1_0) (START ("2010-01-01") END ("2021-12-31") EVERY (INTERVAL 156 day)) DISTRIBUTED BY HASH (c_1_0) BUCKETS 3 properties("replication_num"="3") ;
INSERT INTO t0 VALUES ('2010-01-17 08:00:14'), ('2010-01-22 08:00:14'), (NULL);
INSERT INTO t1 VALUES ('2010-01-01 08:00:14');
CREATE MATERIALIZED VIEW v0 AS SELECT t0_1297.c_0_0 FROM t0 AS t0_1297;
CREATE MATERIALIZED VIEW v0 AS SELECT t0_1299.c_0_0 FROM t0 AS t0_1299;
INSERT INTO t0 VALUES (NULL), ('2010-01-13 08:00:14');
INSERT INTO t1 VALUES ('2010-01-26 08:00:14'), ('2010-01-27 08:00:14'), ('2010-01-23 08:00:14');
INSERT INTO t1 VALUES ('2010-01-27 08:00:14'), ('2010-01-14 08:00:14');
INSERT INTO t1 VALUES ('2010-01-15 08:00:14'), ('2010-01-14 08:00:14'), ('2010-01-01 08:00:14');
INSERT INTO t1 VALUES ('2010-01-01 08:00:14'), ('2010-01-05 08:00:14'), ('2010-01-24 08:00:14'), (NULL);
INSERT INTO t0 VALUES ('2010-01-10 08:00:14'), ('2010-01-21 08:00:14');
INSERT INTO t1 VALUES ('2010-01-27 08:00:14'), ('2010-01-14 08:00:14');
INSERT INTO t0 VALUES ('2010-01-28 08:00:14'), ('2010-01-20 08:00:14');
INSERT INTO t0 VALUES (NULL), (NULL);
INSERT INTO t0 VALUES ('2010-01-03 08:00:14'), ('2010-01-21 08:00:14'), ('2010-01-28 08:00:14');
INSERT INTO t0 VALUES ('2010-01-07 08:00:14'), ('2010-01-22 08:00:14'), ('2010-01-17 08:00:14');
INSERT INTO t0 VALUES ('2010-01-18 08:00:14'), ('2010-01-29 08:00:14');
INSERT INTO t0 VALUES (NULL), (NULL);
DELETE FROM t0 WHERE c_0_0='1970-01-24 20:23:59';
INSERT INTO t0 VALUES ('2010-01-31 08:00:14');
INSERT INTO t0 VALUES ('2009-12-31 08:00:14'), ('2010-01-18 08:00:14');
INSERT INTO t1 VALUES ('2010-01-23 08:00:14'), ('2009-12-31 08:00:14');
INSERT INTO t0 VALUES ('2010-01-30 08:00:14'), (NULL), ('2010-01-10 08:00:14');
CREATE VIEW v67 AS SELECT subt1.c_1_0 FROM  (SELECT t1_1304.c_1_0 FROM t1 AS t1_1304) subt1 FULL JOIN (SELECT t0_1303.c_0_0 FROM t0 AS t0_1303) subt0 ON subt1.c_1_0 = subt0.c_0_0 AND subt1.c_1_0 <= subt0.c_0_0 AND subt1.c_1_0 = subt0.c_0_0 AND subt1.c_1_0 < subt0.c_0_0;
CREATE VIEW v0 AS SELECT subt1.c_1_0 FROM  (SELECT t1_1304.c_1_0 FROM t1 AS t1_1304) subt1 FULL JOIN (SELECT t0_1303.c_0_0 FROM t0 AS t0_1303) subt0 ON subt1.c_1_0 = subt0.c_0_0 AND subt1.c_1_0 <= subt0.c_0_0 AND subt1.c_1_0 = subt0.c_0_0 AND subt1.c_1_0 < subt0.c_0_0 WHERE CAST('1970-01-24 20:23:59' AS BOOLEAN )  ORDER BY subt1.c_1_0 DESC, subt1.c_1_0 ASC;
CREATE VIEW v1 AS SELECT subt1.c_1_0 FROM  (SELECT t1_1304.c_1_0 FROM t1 AS t1_1304) subt1 FULL JOIN (SELECT t0_1303.c_0_0 FROM t0 AS t0_1303) subt0 ON subt1.c_1_0 = subt0.c_0_0 AND subt1.c_1_0 <= subt0.c_0_0 AND subt1.c_1_0 = subt0.c_0_0 AND subt1.c_1_0 < subt0.c_0_0 WHERE (NOT (CAST('1970-01-24 20:23:59' AS BOOLEAN ) )) ORDER BY subt1.c_1_0 DESC, subt1.c_1_0 ASC;
CREATE VIEW v2 AS SELECT subt1.c_1_0 FROM  (SELECT t1_1304.c_1_0 FROM t1 AS t1_1304) subt1 FULL JOIN (SELECT t0_1303.c_0_0 FROM t0 AS t0_1303) subt0 ON subt1.c_1_0 = subt0.c_0_0 AND subt1.c_1_0 <= subt0.c_0_0 AND subt1.c_1_0 = subt0.c_0_0 AND subt1.c_1_0 < subt0.c_0_0 WHERE ((CAST('1970-01-24 20:23:59' AS BOOLEAN ) ) IS NULL) ORDER BY subt1.c_1_0 DESC, subt1.c_1_0 ASC;
CREATE VIEW v3 AS (SELECT subt1.c_1_0 FROM  (SELECT t1_1304.c_1_0 FROM t1 AS t1_1304) subt1 FULL JOIN (SELECT t0_1303.c_0_0 FROM t0 AS t0_1303) subt0 ON subt1.c_1_0 = subt0.c_0_0 AND subt1.c_1_0 <= subt0.c_0_0 AND subt1.c_1_0 = subt0.c_0_0 AND subt1.c_1_0 < subt0.c_0_0 WHERE CAST('1970-01-24 20:23:59' AS BOOLEAN )  ORDER BY subt1.c_1_0 DESC, subt1.c_1_0 ASC) UNION ALL (SELECT subt1.c_1_0 FROM  (SELECT t1_1304.c_1_0 FROM t1 AS t1_1304) subt1 FULL JOIN (SELECT t0_1303.c_0_0 FROM t0 AS t0_1303) subt0 ON subt1.c_1_0 = subt0.c_0_0 AND subt1.c_1_0 <= subt0.c_0_0 AND subt1.c_1_0 = subt0.c_0_0 AND subt1.c_1_0 < subt0.c_0_0 WHERE (NOT (CAST('1970-01-24 20:23:59' AS BOOLEAN ) )) ORDER BY subt1.c_1_0 DESC, subt1.c_1_0 ASC) UNION ALL (SELECT subt1.c_1_0 FROM  (SELECT t1_1304.c_1_0 FROM t1 AS t1_1304) subt1 FULL JOIN (SELECT t0_1303.c_0_0 FROM t0 AS t0_1303) subt0 ON subt1.c_1_0 = subt0.c_0_0 AND subt1.c_1_0 <= subt0.c_0_0 AND subt1.c_1_0 = subt0.c_0_0 AND subt1.c_1_0 < subt0.c_0_0 WHERE ((CAST('1970-01-24 20:23:59' AS BOOLEAN ) ) IS NULL) ORDER BY subt1.c_1_0 DESC, subt1.c_1_0 ASC);
CREATE VIEW v6 AS SELECT v0_1325.c_1_0 FROM v0 AS v0_1325, v2 AS v2_1327, v1 AS v1_1326, t0 AS t0_1323;
CREATE VIEW v4 AS SELECT v0_1325.c_1_0 FROM v0 AS v0_1325, v2 AS v2_1327, v1 AS v1_1326, t0 AS t0_1323 WHERE true;
CREATE VIEW v5 AS SELECT v0_1325.c_1_0 FROM v0 AS v0_1325, v2 AS v2_1327, v1 AS v1_1326, t0 AS t0_1323 WHERE (NOT (true));
CREATE VIEW v7 AS SELECT v0_1325.c_1_0 FROM v0 AS v0_1325, v2 AS v2_1327, v1 AS v1_1326, t0 AS t0_1323 WHERE ((true) IS NULL);
CREATE VIEW v8 AS (SELECT v0_1325.c_1_0 FROM v0 AS v0_1325, v2 AS v2_1327, v1 AS v1_1326, t0 AS t0_1323 WHERE true) UNION ALL (SELECT v0_1325.c_1_0 FROM v0 AS v0_1325, v2 AS v2_1327, v1 AS v1_1326, t0 AS t0_1323 WHERE (NOT (true))) UNION ALL (SELECT v0_1325.c_1_0 FROM v0 AS v0_1325, v2 AS v2_1327, v1 AS v1_1326, t0 AS t0_1323 WHERE ((true) IS NULL));
CREATE VIEW v9 AS SELECT v67_1377.c_1_0 FROM v0 AS v0_1370,  v67 AS v67_1377 INNER JOIN (SELECT t1_1369.c_1_0 FROM t1 AS t1_1369 WHERE true) subt1 ON v67_1377.c_1_0 = subt1.c_1_0 AND v67_1377.c_1_0 >= subt1.c_1_0 AND v67_1377.c_1_0 = subt1.c_1_0 AND v67_1377.c_1_0 != subt1.c_1_0;
CREATE VIEW v10 AS SELECT v67_1377.c_1_0 FROM v0 AS v0_1370,  v67 AS v67_1377 INNER JOIN (SELECT t1_1369.c_1_0 FROM t1 AS t1_1369 WHERE true) subt1 ON v67_1377.c_1_0 = subt1.c_1_0 AND v67_1377.c_1_0 >= subt1.c_1_0 AND v67_1377.c_1_0 = subt1.c_1_0 AND v67_1377.c_1_0 != subt1.c_1_0 WHERE true;
CREATE VIEW v11 AS SELECT v67_1377.c_1_0 FROM v0 AS v0_1370,  v67 AS v67_1377 INNER JOIN (SELECT t1_1369.c_1_0 FROM t1 AS t1_1369 WHERE true) subt1 ON v67_1377.c_1_0 = subt1.c_1_0 AND v67_1377.c_1_0 >= subt1.c_1_0 AND v67_1377.c_1_0 = subt1.c_1_0 AND v67_1377.c_1_0 != subt1.c_1_0 WHERE (NOT (true));
CREATE VIEW v12 AS SELECT v67_1377.c_1_0 FROM v0 AS v0_1370,  v67 AS v67_1377 INNER JOIN (SELECT t1_1369.c_1_0 FROM t1 AS t1_1369 WHERE true) subt1 ON v67_1377.c_1_0 = subt1.c_1_0 AND v67_1377.c_1_0 >= subt1.c_1_0 AND v67_1377.c_1_0 = subt1.c_1_0 AND v67_1377.c_1_0 != subt1.c_1_0 WHERE ((true) IS NULL);
CREATE VIEW v13 AS (SELECT v67_1377.c_1_0 FROM v0 AS v0_1370,  v67 AS v67_1377 INNER JOIN (SELECT t1_1369.c_1_0 FROM t1 AS t1_1369 WHERE true) subt1 ON v67_1377.c_1_0 = subt1.c_1_0 AND v67_1377.c_1_0 >= subt1.c_1_0 AND v67_1377.c_1_0 = subt1.c_1_0 AND v67_1377.c_1_0 != subt1.c_1_0 WHERE true) UNION ALL (SELECT v67_1377.c_1_0 FROM v0 AS v0_1370,  v67 AS v67_1377 INNER JOIN (SELECT t1_1369.c_1_0 FROM t1 AS t1_1369 WHERE true) subt1 ON v67_1377.c_1_0 = subt1.c_1_0 AND v67_1377.c_1_0 >= subt1.c_1_0 AND v67_1377.c_1_0 = subt1.c_1_0 AND v67_1377.c_1_0 != subt1.c_1_0 WHERE (NOT (true))) UNION ALL (SELECT v67_1377.c_1_0 FROM v0 AS v0_1370,  v67 AS v67_1377 INNER JOIN (SELECT t1_1369.c_1_0 FROM t1 AS t1_1369 WHERE true) subt1 ON v67_1377.c_1_0 = subt1.c_1_0 AND v67_1377.c_1_0 >= subt1.c_1_0 AND v67_1377.c_1_0 = subt1.c_1_0 AND v67_1377.c_1_0 != subt1.c_1_0 WHERE ((true) IS NULL));
CREATE VIEW v14 AS SELECT v8_1453.c_1_0 FROM v1 AS v1_1441, t0 AS t0_1438, v3 AS v3_1447, v67 AS v67_1451, v13 AS v13_1445, v5 AS v5_1449, v9 AS v9_1454, v10 AS v10_1442, v4 AS v4_1448, v11 AS v11_1443, t1 AS t1_1439,  v8 AS v8_1453 INNER JOIN (SELECT v7_1452.c_1_0 FROM v7 AS v7_1452 WHERE false ORDER BY EXISTS (SELECT v7_1452.c_1_0 FROM v13 AS v13_1445, v11 AS v11_1443, v6 AS v6_1450, t1 AS t1_1439, v7 AS v7_1452, t0 AS t0_1438, v3 AS v3_1447, v0 AS v0_1440, v1 AS v1_1441, v12 AS v12_1444, v2 AS v2_1446, v4 AS v4_1448, v9 AS v9_1454, v67 AS v67_1451 WHERE ((v7_1452.c_1_0) IS NULL)) ) subv7 ON v8_1453.c_1_0 = subv7.c_1_0 AND v8_1453.c_1_0 = subv7.c_1_0;
CREATE VIEW v15 AS SELECT v8_1453.c_1_0 FROM v1 AS v1_1441, t0 AS t0_1438, v3 AS v3_1447, v67 AS v67_1451, v13 AS v13_1445, v5 AS v5_1449, v9 AS v9_1454, v10 AS v10_1442, v4 AS v4_1448, v11 AS v11_1443, t1 AS t1_1439,  v8 AS v8_1453 INNER JOIN (SELECT v7_1452.c_1_0 FROM v7 AS v7_1452 WHERE false ORDER BY EXISTS (SELECT v7_1452.c_1_0 FROM v13 AS v13_1445, v11 AS v11_1443, v6 AS v6_1450, t1 AS t1_1439, v7 AS v7_1452, t0 AS t0_1438, v3 AS v3_1447, v0 AS v0_1440, v1 AS v1_1441, v12 AS v12_1444, v2 AS v2_1446, v4 AS v4_1448, v9 AS v9_1454, v67 AS v67_1451 WHERE ((v7_1452.c_1_0) IS NULL)) ) subv7 ON v8_1453.c_1_0 = subv7.c_1_0 AND v8_1453.c_1_0 = subv7.c_1_0 WHERE BITMAP_HAS_ANY(bitmap_hash('CNFwzcZA'), BITMAP_INTERSECT(bitmap_hash('O')) ) ;
CREATE VIEW v15 AS SELECT v67_1487.c_1_0 FROM v67 AS v67_1487, t1 AS t1_1474, v2 AS v2_1482, v1 AS v1_1476, v5 AS v5_1485, v10 AS v10_1477, t0 AS t0_1473, v3 AS v3_1483, v4 AS v4_1484, v11 AS v11_1478, v9 AS v9_1490, v14 AS v14_1481, v12 AS v12_1479, v6 AS v6_1486, v0 AS v0_1475;
CREATE VIEW v16 AS SELECT v67_1487.c_1_0 FROM v67 AS v67_1487, t1 AS t1_1474, v2 AS v2_1482, v1 AS v1_1476, v5 AS v5_1485, v10 AS v10_1477, t0 AS t0_1473, v3 AS v3_1483, v4 AS v4_1484, v11 AS v11_1478, v9 AS v9_1490, v14 AS v14_1481, v12 AS v12_1479, v6 AS v6_1486, v0 AS v0_1475 WHERE (NOT ( ( NULL ) = ( v11_1478.c_1_0 ) ));
CREATE VIEW v85 AS SELECT v67_1487.c_1_0 FROM v67 AS v67_1487, t1 AS t1_1474, v2 AS v2_1482, v1 AS v1_1476, v5 AS v5_1485, v10 AS v10_1477, t0 AS t0_1473, v3 AS v3_1483, v4 AS v4_1484, v11 AS v11_1478, v9 AS v9_1490, v14 AS v14_1481, v12 AS v12_1479, v6 AS v6_1486, v0 AS v0_1475 WHERE (NOT ((NOT ( ( NULL ) = ( v11_1478.c_1_0 ) ))));
CREATE VIEW v17 AS SELECT v67_1487.c_1_0 FROM v67 AS v67_1487, t1 AS t1_1474, v2 AS v2_1482, v1 AS v1_1476, v5 AS v5_1485, v10 AS v10_1477, t0 AS t0_1473, v3 AS v3_1483, v4 AS v4_1484, v11 AS v11_1478, v9 AS v9_1490, v14 AS v14_1481, v12 AS v12_1479, v6 AS v6_1486, v0 AS v0_1475 WHERE (((NOT ( ( NULL ) = ( v11_1478.c_1_0 ) ))) IS NULL);
CREATE VIEW v18 AS (SELECT v67_1487.c_1_0 FROM v67 AS v67_1487, t1 AS t1_1474, v2 AS v2_1482, v1 AS v1_1476, v5 AS v5_1485, v10 AS v10_1477, t0 AS t0_1473, v3 AS v3_1483, v4 AS v4_1484, v11 AS v11_1478, v9 AS v9_1490, v14 AS v14_1481, v12 AS v12_1479, v6 AS v6_1486, v0 AS v0_1475 WHERE (NOT ( ( NULL ) = ( v11_1478.c_1_0 ) ))) UNION ALL (SELECT v67_1487.c_1_0 FROM v67 AS v67_1487, t1 AS t1_1474, v2 AS v2_1482, v1 AS v1_1476, v5 AS v5_1485, v10 AS v10_1477, t0 AS t0_1473, v3 AS v3_1483, v4 AS v4_1484, v11 AS v11_1478, v9 AS v9_1490, v14 AS v14_1481, v12 AS v12_1479, v6 AS v6_1486, v0 AS v0_1475 WHERE (NOT ((NOT ( ( NULL ) = ( v11_1478.c_1_0 ) ))))) UNION ALL (SELECT v67_1487.c_1_0 FROM v67 AS v67_1487, t1 AS t1_1474, v2 AS v2_1482, v1 AS v1_1476, v5 AS v5_1485, v10 AS v10_1477, t0 AS t0_1473, v3 AS v3_1483, v4 AS v4_1484, v11 AS v11_1478, v9 AS v9_1490, v14 AS v14_1481, v12 AS v12_1479, v6 AS v6_1486, v0 AS v0_1475 WHERE (((NOT ( ( NULL ) = ( v11_1478.c_1_0 ) ))) IS NULL));
CREATE VIEW v19 AS SELECT v6_1590.c_1_0 FROM v6 AS v6_1590, t0 AS t0_1573, v17 AS v17_1584, v18 AS v18_1585, v9 AS v9_1595, v67 AS v67_1591, v11 AS v11_1578, v14 AS v14_1581, v0 AS v0_1575, v1 AS v1_1576, v12 AS v12_1579, v8 AS v8_1593, v3 AS v3_1587, v13 AS v13_1580, v85 AS v85_1594, v15 AS v15_1582, v2 AS v2_1586, v4 AS v4_1588, v10 AS v10_1577, v5 AS v5_1589, t1 AS t1_1574, v7 AS v7_1592, v16 AS v16_1583;
CREATE VIEW v20 AS SELECT v6_1590.c_1_0 FROM v6 AS v6_1590, t0 AS t0_1573, v17 AS v17_1584, v18 AS v18_1585, v9 AS v9_1595, v67 AS v67_1591, v11 AS v11_1578, v14 AS v14_1581, v0 AS v0_1575, v1 AS v1_1576, v12 AS v12_1579, v8 AS v8_1593, v3 AS v3_1587, v13 AS v13_1580, v85 AS v85_1594, v15 AS v15_1582, v2 AS v2_1586, v4 AS v4_1588, v10 AS v10_1577, v5 AS v5_1589, t1 AS t1_1574, v7 AS v7_1592, v16 AS v16_1583 WHERE JSON_EXISTS(parse_json('{"b": true, "OCX": 556397652, "QRA6to6uM": {"0": null, "mn": null, "Q60o": "", "qB": false, "gI": 0.6385858944442148, "5ylhM5Z": -1986434448, "oe": 1237477003, "I": true, "0Cn": false, "l": {"NcX8": true, "e": null, "eFU": true, "G": 0.12398133336901751, "W": 0.2733151721174196, "Vc4i14": null, "oSm": 0.30127025953337483, "VhO": null, "bIG4lE": true, "o": 0.12020896641366874, "O": true}}, "xVD": null, "H": -808488549, "dt": 0.04646979454612199, "iS8o": [0.2946820154366889, [], null, 1330077497, -1999935246], "nM1": "0z5o9I", "t": {"0": {"q": "7KjSut", "GdB": null, "vG96TKKe": null, "S9n": null, "Z": -352899787, "Y6": ""}, "CS": null, "s4UBRbLvd": true, "B": {}, "e": ["6tu", 1330077497, -1986434448, "k", 0.6445984876288092, 0.6385858944442148, null, null], "E": null, "f": 618333501, "w": "9Ni", "a6zyRNH": -352899787, "9": -1986434448, "KRthf": 1330077497}, "oJ0": "", "T": ["Oe1"], "EOpGWr": 0.6445984876288092, "7": null, "czIe0XBUw": -1790928235}'), (((('')||('')))||(RTRIM('6u') ))) ;
CREATE VIEW v44 AS SELECT v6_1590.c_1_0 FROM v6 AS v6_1590, t0 AS t0_1573, v17 AS v17_1584, v18 AS v18_1585, v9 AS v9_1595, v67 AS v67_1591, v11 AS v11_1578, v14 AS v14_1581, v0 AS v0_1575, v1 AS v1_1576, v12 AS v12_1579, v8 AS v8_1593, v3 AS v3_1587, v13 AS v13_1580, v85 AS v85_1594, v15 AS v15_1582, v2 AS v2_1586, v4 AS v4_1588, v10 AS v10_1577, v5 AS v5_1589, t1 AS t1_1574, v7 AS v7_1592, v16 AS v16_1583 WHERE (NOT (JSON_EXISTS(parse_json('{"b": true, "OCX": 556397652, "QRA6to6uM": {"0": null, "mn": null, "Q60o": "", "qB": false, "gI": 0.6385858944442148, "5ylhM5Z": -1986434448, "oe": 1237477003, "I": true, "0Cn": false, "l": {"NcX8": true, "e": null, "eFU": true, "G": 0.12398133336901751, "W": 0.2733151721174196, "Vc4i14": null, "oSm": 0.30127025953337483, "VhO": null, "bIG4lE": true, "o": 0.12020896641366874, "O": true}}, "xVD": null, "H": -808488549, "dt": 0.04646979454612199, "iS8o": [0.2946820154366889, [], null, 1330077497, -1999935246], "nM1": "0z5o9I", "t": {"0": {"q": "7KjSut", "GdB": null, "vG96TKKe": null, "S9n": null, "Z": -352899787, "Y6": ""}, "CS": null, "s4UBRbLvd": true, "B": {}, "e": ["6tu", 1330077497, -1986434448, "k", 0.6445984876288092, 0.6385858944442148, null, null], "E": null, "f": 618333501, "w": "9Ni", "a6zyRNH": -352899787, "9": -1986434448, "KRthf": 1330077497}, "oJ0": "", "T": ["Oe1"], "EOpGWr": 0.6445984876288092, "7": null, "czIe0XBUw": -1790928235}'), (((('')||('')))||(RTRIM('6u') ))) ));
CREATE VIEW v21 AS SELECT v6_1590.c_1_0 FROM v6 AS v6_1590, t0 AS t0_1573, v17 AS v17_1584, v18 AS v18_1585, v9 AS v9_1595, v67 AS v67_1591, v11 AS v11_1578, v14 AS v14_1581, v0 AS v0_1575, v1 AS v1_1576, v12 AS v12_1579, v8 AS v8_1593, v3 AS v3_1587, v13 AS v13_1580, v85 AS v85_1594, v15 AS v15_1582, v2 AS v2_1586, v4 AS v4_1588, v10 AS v10_1577, v5 AS v5_1589, t1 AS t1_1574, v7 AS v7_1592, v16 AS v16_1583 WHERE ((JSON_EXISTS(parse_json('{"b": true, "OCX": 556397652, "QRA6to6uM": {"0": null, "mn": null, "Q60o": "", "qB": false, "gI": 0.6385858944442148, "5ylhM5Z": -1986434448, "oe": 1237477003, "I": true, "0Cn": false, "l": {"NcX8": true, "e": null, "eFU": true, "G": 0.12398133336901751, "W": 0.2733151721174196, "Vc4i14": null, "oSm": 0.30127025953337483, "VhO": null, "bIG4lE": true, "o": 0.12020896641366874, "O": true}}, "xVD": null, "H": -808488549, "dt": 0.04646979454612199, "iS8o": [0.2946820154366889, [], null, 1330077497, -1999935246], "nM1": "0z5o9I", "t": {"0": {"q": "7KjSut", "GdB": null, "vG96TKKe": null, "S9n": null, "Z": -352899787, "Y6": ""}, "CS": null, "s4UBRbLvd": true, "B": {}, "e": ["6tu", 1330077497, -1986434448, "k", 0.6445984876288092, 0.6385858944442148, null, null], "E": null, "f": 618333501, "w": "9Ni", "a6zyRNH": -352899787, "9": -1986434448, "KRthf": 1330077497}, "oJ0": "", "T": ["Oe1"], "EOpGWr": 0.6445984876288092, "7": null, "czIe0XBUw": -1790928235}'), (((('')||('')))||(RTRIM('6u') ))) ) IS NULL);

--- a/fe/fe-core/src/main/java/com/starrocks/sql/optimizer/Memo.java
+++ b/fe/fe-core/src/main/java/com/starrocks/sql/optimizer/Memo.java
@@ -295,20 +295,24 @@ public class Memo {
     }
 
     /*
-     * remove unreachable group, and rebuild all group expression.
-     * only used in logical rewrite phase, logical expression must only one in a group,
-     * and remove groupExpression one by one is too slow, so rebuild directly
+     * @Note: The function only work in logical rewrite phase !!!
+     *
+     * When performing replaceRewriteExpression, some groups may not be reachable by rootGroup.
+     * These groups should be replaced.
+     * In order to reduce the number of groups entering Memo,
+     * we will delete inaccessible groups in this function.
      */
-    public void rebuildGroupExpressions() {
+    public void removeUnreachableGroup() {
         LinkedList<Integer> touch = new LinkedList<>();
         touch.add(rootGroup.getId());
         deepSearchGroup(rootGroup, touch);
         groups.removeIf(g -> !touch.contains(g.getId()));
         groupExpressions.clear();
 
+        // only used in logical rewrite phase, logical expression must only one in a group,
+        // and remove groupExpression one by one is too slow, so rebuild directly
         for (Group group : groups) {
             group.getLogicalExpressions().forEach(l -> groupExpressions.put(l, l));
-            group.getPhysicalExpressions().forEach(p -> groupExpressions.put(p, p));
         }
     }
 

--- a/fe/fe-core/src/main/java/com/starrocks/sql/optimizer/Memo.java
+++ b/fe/fe-core/src/main/java/com/starrocks/sql/optimizer/Memo.java
@@ -294,6 +294,11 @@ public class Memo {
         }
     }
 
+    /*
+     * remove unreachable group, and rebuild all group expression.
+     * only used in logical rewrite phase, logical expression must only one in a group,
+     * and remove groupExpression one by one is too slow, so rebuild directly
+     */
     public void rebuildGroupExpressions() {
         LinkedList<Integer> touch = new LinkedList<>();
         touch.add(rootGroup.getId());

--- a/fe/fe-core/src/main/java/com/starrocks/sql/optimizer/Memo.java
+++ b/fe/fe-core/src/main/java/com/starrocks/sql/optimizer/Memo.java
@@ -10,7 +10,6 @@ import com.starrocks.common.Pair;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 
-import java.util.ArrayList;
 import java.util.Iterator;
 import java.util.LinkedList;
 import java.util.List;
@@ -295,22 +294,16 @@ public class Memo {
         }
     }
 
-    /**
-     * When performing replaceRewriteExpression, some groups may not be reachable by rootGroup.
-     * These groups should be replaced.
-     * In order to reduce the number of groups entering Memo,
-     * we will delete inaccessible groups in this function.
-     */
-    public void removeUnreachableGroup() {
+    public void rebuildGroupExpressions() {
         LinkedList<Integer> touch = new LinkedList<>();
         touch.add(rootGroup.getId());
         deepSearchGroup(rootGroup, touch);
+        groups.removeIf(g -> !touch.contains(g.getId()));
+        groupExpressions.clear();
 
-        List<Group> groupsCopy = new ArrayList<>(groups);
-        for (Group group : groupsCopy) {
-            if (!touch.contains(group.getId())) {
-                removeOneGroup(group);
-            }
+        for (Group group : groups) {
+            group.getLogicalExpressions().forEach(l -> groupExpressions.put(l, l));
+            group.getPhysicalExpressions().forEach(p -> groupExpressions.put(p, p));
         }
     }
 

--- a/fe/fe-core/src/main/java/com/starrocks/sql/optimizer/Optimizer.java
+++ b/fe/fe-core/src/main/java/com/starrocks/sql/optimizer/Optimizer.java
@@ -280,7 +280,7 @@ public class Optimizer {
     private void cleanUpMemoGroup(Memo memo) {
         // Rewrite maybe produce empty groups, we need to remove them.
         memo.removeAllEmptyGroup();
-        memo.rebuildGroupExpressions();
+        memo.removeUnreachableGroup();
     }
 
     /**

--- a/fe/fe-core/src/main/java/com/starrocks/sql/optimizer/Optimizer.java
+++ b/fe/fe-core/src/main/java/com/starrocks/sql/optimizer/Optimizer.java
@@ -204,15 +204,15 @@ public class Optimizer {
         // because of the Filter node needs to be merged first to avoid the Limit node
         // cannot merge
         ruleRewriteIterative(memo, rootTaskContext, RuleSetType.PUSH_DOWN_PREDICATE);
-        ruleRewriteIterative(memo, rootTaskContext, new MergeTwoProjectRule());
-        ruleRewriteOnlyOnce(memo, rootTaskContext, new PushDownAggToMetaScanRule());
-
         cleanUpMemoGroup(memo);
 
+        ruleRewriteIterative(memo, rootTaskContext, new MergeTwoProjectRule());
+        ruleRewriteOnlyOnce(memo, rootTaskContext, new PushDownAggToMetaScanRule());
         ruleRewriteOnlyOnce(memo, rootTaskContext, new PushDownPredicateWindowRankRule());
         ruleRewriteOnlyOnce(memo, rootTaskContext, new PushDownJoinOnExpressionToChildProject());
         ruleRewriteOnlyOnce(memo, rootTaskContext, RuleSetType.PRUNE_COLUMNS);
-
+        cleanUpMemoGroup(memo);
+        
         // After prune columns, the output column in the logical property may outdated, because of the following rule
         // will use the output column, we need to derive the logical property here.
         memo.deriveAllGroupLogicalProperty();
@@ -280,7 +280,7 @@ public class Optimizer {
     private void cleanUpMemoGroup(Memo memo) {
         // Rewrite maybe produce empty groups, we need to remove them.
         memo.removeAllEmptyGroup();
-        memo.removeUnreachableGroup();
+        memo.rebuildGroupExpressions();
     }
 
     /**

